### ISSUE TITLE
Simplifying Medicine: Developing an Accessible Health Information Platform

### ISSUE THREAD

**Issue by Kshah002** [SrijanShovit/HealthLearning#134]
### Is your feature request related to a problem? Please describe.

Not everyone is familiar with all diseases and the medical terminology used by professionals. Therefore, an interface should be created to address this issue.

### Describe the solution you'd like

To develop an LLM model trained on a dataset encompassing various medical terms and diseases, the fundamental concept is for users to inquire about specific diseases and receive relevant, accurate responses. I aim to leverage a pre-trained model, such as LLaMA 2, using Hugging Face.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Kshah002** [comment]
dataset that can be used - https://huggingface.co/datasets/gamino/wiki_medical_terms

**Comment by Kshah002** [comment]
hi @SrijanShovit I have raised the issue and have worked with similar project using LLama. So I would request to add the required labels and assign the task to me

**Comment by SrijanShovit** [comment]
Hmm....that looks cool. What are your detailed steps?

**Comment by Kshah002** [comment]
I wanted to try thus have already started working on the project  so here is a general idea of how i am to finetune the llama2 model -

First step was to get dataset according to llama2 format however i got the formatted dataset from huggingface itself - https://huggingface.co/datasets/aboonaji/wiki_medical_terms_llam2_format
Using 4-bit quantization while loading the pre-trained LLAma2 model - https://huggingface.co/aboonaji/llama2finetune-v2
To use 4-bit weights, with float16 for computation, and specifying the quantization type as nf4
Loading a tokenizer compatible with the LLaMA model, setting the pad token 
Will be using peft for fine tuning

I am using google colab so, the parameters are considered keeping that in mind. I hope you get the gist of what I am trying to do

**Comment by Kshah002** [comment]
@SrijanShovit Hii there. Any updates ??

**Comment by SrijanShovit** [comment]
Yes looks good. Do make documentation along with your minor code steps and keep committing in a single PR.

**Comment by Kshah002** [comment]
Hey @SrijanShovit can you help me reopen the issue ? I am done with my task and was about to upload and saw the issue is closed. Can you help me out here ? Sorry for being a bit late though.

Or should I raise a new issue ?

**Comment by Kshah002** [comment]
Hi @SrijanShovit any updates ??
